MAVProxy/modules/lib/ntrip.py

```diff
@@ -121,7 +123,8 @@ def get_ID(self):
 
     def read(self):
         if self.socket is None:
-            time.sleep(0.1)
+            if self.socket_pending is None:
+                time.sleep(0.1)
```

Comment:
This structure will introduce a 100ms sleep every other call.

Scribbling down a `last_connect_attempt = time.time()` and using it to determine when to try again would avoid this.

Reply:
fixed